actor/src/main/scala/org/apache/pekko/actor/LightArrayRevolverScheduler.scala

```diff
@@ -105,12 +105,14 @@ class LightArrayRevolverScheduler(config: Config, log: LoggingAdapter, threadFac
 
   override def scheduleWithFixedDelay(initialDelay: FiniteDuration, delay: FiniteDuration)(runnable: Runnable)(
       implicit executor: ExecutionContext): Cancellable = {
+    checkZeroPeriod(delay.toNanos)
     checkMaxDelay(roundUp(delay).toNanos)
     super.scheduleWithFixedDelay(initialDelay, delay)(runnable)
   }
 
   override def schedule(initialDelay: FiniteDuration, delay: FiniteDuration, runnable: Runnable)(
       implicit executor: ExecutionContext): Cancellable = {
+    checkZeroPeriod(delay.toNanos)
```

Comment:
I think we should add the check to `scheduleAtFixedRate` and `scheduleWithFixedDelay`.

And inline the method,otherwise it will add a stackframe

Reply:
make sense,  this is not something that wroth to abstract, use inline we could gain less stack frame

Reply:
There is another viewpoint on this: "It is fine because it will always be at the top of the stack frame." I have made an example to verify it, see the  another comment on below.